src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -640,12 +651,23 @@ impl<'tcx> Tree {
         );
         // Register new_tag as a child of parent_tag
         self.nodes.get_mut(parent_idx).unwrap().children.push(idx);
-        // Initialize perms
-        let perm = LocationState::new_init(default_initial_perm, strongest_idempotent);
-        for (_perms_range, perms) in self.rperms.iter_mut(reborrow_range.start, reborrow_range.size)
-        {
-            perms.insert(idx, perm);
-        }
+
+        let NewPermission { freeze_perm, nonfreeze_perm, .. } = perm;
+        this.visit_freeze_sensitive(place, size, |mut range, frozen| {
```

Comment:
And here we have a third call to `visit_freeze_sensitive`? I don't understand the structure of this, but it's definitely not the structure I was hoping for. Check out Stacked Borrows for a template.

Reply:
Putting this part into the `visit_freeze_sensitive` call in
`new_child` seems to work (all tests pass), but I don't know if this
will interfere with the `update_last_accessed_after_retag` call.  I
will look more into this.